File: Regression/EP_Web/Links/02_page_links.rb
#Remove All instances of Current Page from spreadsheet
require 'rubygems'
require 'watir-webdriver'
require "win32ole"

excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\EP_Web\\Links\\Links.xlsx")
worksheet = workbook.worksheets(2) 
browser = Watir::Browser.new :firefox

rows = 2		#2
while    
rows <= 64		#63

page=worksheet.cells(rows,"A").value
xpath=worksheet.cells(rows,"B").value
expected=worksheet.cells(rows,"C").value

string_from_excel = "goto"
browser.send string_from_excel, page
browser.link(:xpath => xpath).click
sleep 1

if
browser.text.include? 'Server Error'
print "Server Error, "
end

if
browser.url == expected
puts "good"
else
puts "bad"
end
rows=rows+1
end

browser.close 

$end
workbook.Save
workbook.Close

$end
sleep 2

File: Regression/EP_Web/Links/02_page_links.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")
    driver = webdriver.Firefox()

workbook = xlrd.open_workbook("./links.xlsx")
worksheet = workbook.sheet_by_name('Page Links')

for current_row in range(1,worksheet.nrows):
    page = worksheet.row(current_row)[0].value
    xpath = worksheet.row(current_row)[1].value
    expected = worksheet.row(current_row)[2].value

    driver.get(page)
    elem = driver.find_element_by_xpath(xpath).click()

    if  "Not Found" in driver.page_source:
        print ("Failed - Not Found")

    if  "Server Error" in driver.page_source:
        print ("Failed - Server Error")

    if driver.current_url == expected:
        print ("Passed - ", expected)
    else:
        print ("Failed - ", expected)

driver.close()